src/utils.ts

```diff
@@ -25,68 +24,100 @@ import { PaymentRequest } from './model/PaymentRequest.js';
 function splitAmount(
 	value: number,
 	keyset: Keys,
-	amountPreference?: Array<AmountPreference>,
-	isDesc?: boolean
+	split?: Array<number>,
+	order?: string
 ): Array<number> {
 	const chunks: Array<number> = [];
-	if (amountPreference) {
-		chunks.push(...getPreference(value, keyset, amountPreference));
+	if (split) {
+		if (split.reduce((a: number, b: number) => a + b, 0) > value) {
+			throw new Error(
+				`Split is greater than total amount: ${split.reduce(
+					(a: number, b: number) => a + b,
+					0
+				)} > ${value}`
+			);
+		}
+		chunks.push(...getPreference(value, keyset, split));
 		value =
 			value -
 			chunks.reduce((curr: number, acc: number) => {
 				return curr + acc;
 			}, 0);
 	}
-	const sortedKeyAmounts: Array<number> = Object.keys(keyset)
-		.map((k) => parseInt(k))
-		.sort((a, b) => b - a);
-	sortedKeyAmounts.forEach((amt) => {
+	const sortedKeyAmounts = getKeysetAmounts(keyset);
+	sortedKeyAmounts.forEach((amt: number) => {
 		const q = Math.floor(value / amt);
 		for (let i = 0; i < q; ++i) chunks.push(amt);
 		value %= amt;
 	});
-	return chunks.sort((a, b) => (isDesc ? b - a : a - b));
+	return chunks.sort((a, b) => (order === 'desc' ? b - a : a - b));
+}
+
+function getKeepAmounts(
```

Comment:
This function seems to not be covered by unit tests. I will not enforce a 100% coverage, but please lmk if this is on purpose.